SpatialGDK/Source/SpatialGDK/Private/Tests/RPCContainerTest.cpp

```diff
@@ -0,0 +1,232 @@
+	RPCs.ProcessRPCs(Delegate);
+
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, ReliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_two_values_of_same_type_have_been_added_and_processed_THEN_nothing_is_queued)
+{
+	UMockObject* TargetObject = NewObject<UMockObject>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params1 = CreateMockParameters(TargetObject, UnreliableType);
+	FPendingRPCParamsPtr Params2 = CreateMockParameters(TargetObject, UnreliableType);
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(Params1), UnreliableType);
+	RPCs.QueueRPC(MoveTemp(Params2), UnreliableType);
+
+	FProcessRPCDelegate Delegate;
+	Delegate.BindUObject(TargetObject, &UMockObject::ProcessRPC);
+	RPCs.ProcessRPCs(Delegate);
+
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, UnreliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_two_values_of_different_type_have_been_added_THEN_they_are_queued)
+{
+	UMockObject* TargetObject = NewObject<UMockObject>();
+
+	FPendingRPCParamsPtr ParamsUnreliable = CreateMockParameters(TargetObject, UnreliableType);
+	FUnrealObjectRef ObjecRefUnreliable = ParamsUnreliable->ObjectRef;
+
+	FPendingRPCParamsPtr ParamsReliable = CreateMockParameters(TargetObject, ReliableType);
+	FUnrealObjectRef ObjecRefReliable = ParamsReliable->ObjectRef;
+
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(ParamsUnreliable), UnreliableType);
+	RPCs.QueueRPC(MoveTemp(ParamsReliable), ReliableType);
+
+	TestTrue("Queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRefUnreliable.Entity, UnreliableType));
+	TestTrue("Queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRefReliable.Entity, ReliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_two_values_of_different_type_have_been_added_and_processed_THEN_nothing_is_queued)
+{
+	UMockObject* TargetObject = NewObject<UMockObject>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr ParamsUnreliable = CreateMockParameters(TargetObject, UnreliableType);
+	FPendingRPCParamsPtr ParamsReliable = CreateMockParameters(TargetObject, ReliableType);
+
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(ParamsUnreliable), UnreliableType);
+	RPCs.QueueRPC(MoveTemp(ParamsReliable), ReliableType);
+
+	FProcessRPCDelegate Delegate;
+	Delegate.BindUObject(TargetObject, &UMockObject::ProcessRPC);
+	RPCs.ProcessRPCs(Delegate);
+
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, UnreliableType));
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, ReliableType));
+
+    return true;
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_six_values_of_different_type_have_been_added_and_processed_THEN_they_have_been_processed_in_order)
```

Comment:
Six or four or sixteen?

Reply:
Good catch! That's outdated, will fix it.

Reply:
Is it significant how many? I would have thought its just that the order is respected. In which case remove all reference to numbers from the test name.

Reply:
Good point, the number is not that significant.

Reply:
Removed numbers from test names.